src/mame/misc/orientp.cpp

```diff
@@ -0,0 +1,280 @@
+	kbdc.out_sl_callback().set(FUNC(orientalpearl_state::multiplex_7seg_w));   // select  block of 7seg modules by multiplexing the SL scan lines
+	kbdc.in_rl_callback().set(FUNC(orientalpearl_state::keyboard_r));          // keyboard Return Lines
+	kbdc.out_disp_callback().set(FUNC(orientalpearl_state::display_7seg_data_w));
+	
+	// Video
+	config.set_default_layout(layout_orientp);
+    // sound hardware
+	SPEAKER(config, "mono").front_center();
+	ay8910_device &ay1(AY8910(config, "ay1", XTAL(10'738'635) / 6));
+	ay1.add_route(ALL_OUTPUTS, "mono", 1.0);
+	ay1.port_a_write_callback().set(FUNC(orientalpearl_state::ay1_port_a_w));
+	ay1.port_b_write_callback().set(FUNC(orientalpearl_state::ay1_port_b_w));
+	
+	ym2413_device &opll(YM2413(config, "opll", 3.579545_MHz_XTAL));
+	opll.add_route(ALL_OUTPUTS, "mono", 1.0);
+
+	OKIM6295(config, "oki", XTAL(10'738'000) / 6, okim6295_device::PIN7_HIGH).add_route(ALL_OUTPUTS, "mono", 1.0);  // Clock frequency & pin 7 not verified
+
+}
+
+ROM_START( east8 )
+	ROM_REGION( 0x10000, "maincpu", 0 ) 
+	ROM_LOAD( "27c512.u33", 0x00000, 0x10000,  CRC(85e28db5) SHA1(96f80a7d2214672c09b8f719cb573e77b8bac731) ) // Main program.  EAST8  v1.00 string
+
+    ROM_REGION( 0x1000, "mcu", 0 )
+    ROM_LOAD( "at89s51.u39", 0x0000, 0x1000,   CRC(a55b63a8) SHA1(9ef88bba4a46ccd969d80882e9c36eb2f0c9e4bf) ) //  Microcontroller Protection. 
+ 
+    ROM_REGION( 0x40000, "oki", 0 )
+    ROM_LOAD( "w27c020.bin", 0x00000, 0x40000, CRC(f962ed1c) SHA1(c69cd9619c794e77a0122fc82d36662494ceb0be) ) //  No label  Rom. From Nreat-new GN-001 pcb. misspelled as Great-new?
+
+    ROM_END
+
+ROM_START( east8a )
+	ROM_REGION( 0x10000, "maincpu", 0 ) 
+	ROM_LOAD( "w27c512.u33", 0x00000, 0x10000, CRC(8d3d1e91) SHA1(b80907df0878057a1ded8b56225059e06382b9d6) ) // Main program.  EAST8  v1.05 string
+
+    ROM_REGION( 0x1000, "mcu", 0 )
+    ROM_LOAD( "at89s51.u39", 0x0000, 0x1000, BAD_DUMP CRC(a55b63a8) SHA1(9ef88bba4a46ccd969d80882e9c36eb2f0c9e4bf) ) //  Microcontroller Protection. From the set 2. Marked as Bad dump for now
+ 
+    ROM_REGION( 0x40000, "oki", ROMREGION_ERASE00 )
+    ROM_LOAD( "w27c020.bin", 0x00000, 0x40000, CRC(f962ed1c) BAD_DUMP SHA1(c69cd9619c794e77a0122fc82d36662494ceb0be) ) //  Voices Rom.  From the set 1. Marked as Bad dump for now
+
+    ROM_END
+
+
+
+} // anonymous namespace
+
+
+//   YEAR   NAME    PARENT   MACHINE    INPUT    STATE                INIT        ROT    COMPANY       FULLNAME                 FLAGS
+GAME( 199?, east8,   0,       orientp,   east8, orientalpearl_state, empty_init, ROT0, "<unknown>", "Unknown EAST8 Set 1 (v1.00)",  MACHINE_NOT_WORKING | MACHINE_UNEMULATED_PROTECTION | MACHINE_IMPERFECT_SOUND | MACHINE_MECHANICAL | MACHINE_REQUIRES_ARTWORK )
+GAME( 199?, east8a,  0,       orientp,   east8, orientalpearl_state, empty_init, ROT0, "<unknown>", "Unknown EAST8 Set 2 (v1.05)",  MACHINE_NOT_WORKING | MACHINE_UNEMULATED_PROTECTION | MACHINE_IMPERFECT_SOUND | MACHINE_MECHANICAL | MACHINE_REQUIRES_ARTWORK )
```

Comment:
Please try to follow conventions for descriptions and use mnemonic suffixes for clones.